### PR TITLE
fix: load flickering document synchronously

### PR DESCRIPTION
**Pull request by ukstv**
Maybe works.

### CODE REVIEW

File: packages/core/src/__tests__/ceramic-anchor.test.ts [hunk 1]
```diff
@@ -311,7 +311,7 @@ describe('Ceramic anchoring', () => {
     expect(doctype1.content).toEqual({ x: 6 })
     expect(doctype1.state.log.length).toEqual(8)
 
-    const doctype2 = await ceramic2.loadDocument(doctype1.id)
+    const doctype2 = await ceramic2.loadDocument(doctype1.id, {sync: true})
```
Comment (oed) [inline packages/core/src/__tests__/ceramic-anchor.test.ts:314]: Isn't this the default value for this option?

Comment (ukstv) [reply packages/core/src/__tests__/ceramic-anchor.test.ts:314]: It is indeed.

Comment (oed) [reply packages/core/src/__tests__/ceramic-anchor.test.ts:314]: So does this really solve the problem?

Comment (ukstv) [reply packages/core/src/__tests__/ceramic-anchor.test.ts:314]: Not at all.

**Comment by stbrody** [comment]
I think what we need to do here is something similar to [what we do in the integration tests](https://github.com/3box/ceramic-integration-tests/blob/main/src/utils.ts#L11) for waiting for an anchor.  Basically a loop that waits on change events and then checks some predicate - in this case whether the update has been loaded - repeatedly until a timeout is hit.

**Comment by ukstv** [comment]
Dropped in favour of #960

**Pull request closed without merging**